Core/clim-core/gadgets/menu.lisp

```diff
@@ -249,8 +339,15 @@
                                          :armed-callback 'arm-menu-button-callback
                                          :disarmed-callback 'disarm-menu-button-callback))
                 (rack (make-menu-bar value sub-pane 'vmenu-pane))
-                (border (make-pane 'raised-pane :contents (list rack))))
+                (border (make-pane 'raised-pane :contents (list rack)))
+                (active (remove-if-not (lambda (elt)
+                                         (and (typep elt 'menu-button-pane)
+                                              (gadget-active-p elt)))
+                                       ;; Composite panes adopt children in
+                                       ;; order, so they are in reverse order.
+                                       (reverse (sheet-children rack)))))
```

Comment:
Is it necessary to cache this? Couldn't `menu-children` do this on demand?

Reply:
In particular since the method for frames is identical.

Reply:
OK. I've changed menu-children to a function too and removed submenu-items function.